jme3-core/src/main/java/com/jme3/animation/SkeletonControl.java

```diff
@@ -443,9 +459,19 @@ public Node getAttachmentsNode(String boneName) {
                     + "in the skeleton.");
         }
 
-        Node n = b.getAttachmentsNode();
-        Node model = (Node) spatial;
-        model.attachChild(n);
+        updateTargetsAndMaterials(spatial);
+        Node n = b.getAttachmentsNode(targetGeometry);
```

Comment:
I assume all the animated geometries in the model will have the same global transform, so it shouldn't matter which one I use. If my assumption is wrong, I will change the code as you suggest.